pandas/core/frame.py

```diff
@@ -11069,7 +11069,11 @@ def std(
         numeric_only: bool = False,
         **kwargs,
     ):
-        return super().std(axis, skipna, ddof, numeric_only, **kwargs)
+        result = super().std(axis, skipna, ddof, numeric_only, **kwargs)
+        if isinstance(result, Series):
+            return result.__finalize__(self, method="std")
+        else:
+            return result
```

Comment:
this should be unreachable, and looks like a case where we humans looking at the code can infer things which the type checker can't

I'd suggest
```
result = cast(Series, super().std(...)
return result.__finalize_(...)
```

Reply:
Thank you so much, Mr Gorelli! I appreciate your help so much, I will submit this:)